compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -708,6 +708,27 @@ struct RewriteExternCallOpToDynamicImportCallOp
                                          "and does not need an import wrapper");
     }
 
+    // The call may need some additional internal fields appended.
+    SmallVector<StringRef> extraFields;
+    if (auto extraFieldsAttr =
+            calleeOp->getAttrOfType<ArrayAttr>("hal.import.fields")) {
+      for (auto extraFieldAttr : extraFieldsAttr) {
+        extraFields.push_back(extraFieldAttr.cast<StringAttr>().getValue());
+      }
+    }
+
+    if (calleeOp->hasAttr("hal.import.bitcode")) {
```

Comment:
Maybe `hal.import.cconv` and have it be a string attr defining the calling convention. How the args/results are passed is orthogonal to where the code comes from (bitcode, object files, externally-linked content when generating static libraries, etc) and we want to be able to do this kind of struct packing for all of those. Just picking a name but a calling convention of `"struct"` would be fine to start. Can eventually get fancy and have a few but a placeholder is good enough for now :)

Reply:
I added a enumattr for the calling convention.. For now I just split the bitcode linking patterns to apply separately on the function and call. Ill do the dynamic import ones as a follow up?